wgpu-core/src/instance.rs

```diff
@@ -228,11 +244,60 @@ impl<B: GfxBackend> Adapter<B> {
         Self {
             raw,
             features,
+            private_features,
             limits,
             life_guard: LifeGuard::new("<Adapter>"),
         }
     }
 
+    pub(crate) fn get_texture_format_features(
+        &self,
+        format: wgt::TextureFormat,
+    ) -> wgt::TextureFormatFeatures {
+        let texture_format_properties = self
+            .raw
+            .physical_device
+            .format_properties(Some(conv::map_texture_format(
+                format,
+                self.private_features,
+            )))
+            .optimal_tiling;
+
+        let mut allowed_usages = wgt::TextureUsage::empty();
+        if texture_format_properties.contains(hal::format::ImageFeature::SAMPLED) {
+            allowed_usages |= wgt::TextureUsage::SAMPLED;
+        }
+        if texture_format_properties.contains(hal::format::ImageFeature::STORAGE) {
+            allowed_usages |= wgt::TextureUsage::STORAGE;
+        }
+        if texture_format_properties.contains(hal::format::ImageFeature::COLOR_ATTACHMENT) {
+            allowed_usages |= wgt::TextureUsage::RENDER_ATTACHMENT;
+        }
+        if texture_format_properties.contains(hal::format::ImageFeature::DEPTH_STENCIL_ATTACHMENT) {
+            allowed_usages |= wgt::TextureUsage::RENDER_ATTACHMENT;
+        }
+        if texture_format_properties.contains(hal::format::ImageFeature::BLIT_SRC) {
+            allowed_usages |= wgt::TextureUsage::COPY_SRC;
+        }
+        if texture_format_properties.contains(hal::format::ImageFeature::BLIT_DST) {
+            allowed_usages |= wgt::TextureUsage::COPY_DST;
+        }
+
+        let mut flags = wgt::TextureFormatFeatureFlags::empty();
+        if texture_format_properties.contains(hal::format::ImageFeature::STORAGE_ATOMIC) {
+            flags |= wgt::TextureFormatFeatureFlags::STORAGE_ATOMICS;
+        }
+        // TODO: hal update
+        //if texture_format_properties.contains(hal::format::ImageFeature::STORAGE_READ_WRITE) {
```

Comment:
can now be implemented thanks to you!

Reply:
😊 
once we managed to upgrade to newest gfx that is :)